src/main/webapp/discussion.js

```diff
@@ -103,53 +129,54 @@ async function fetchDiscussion() {
 }
 
 /**
- * Creates an element for displaying {@code comment}.
+ * Renders a comment and its replies, with a form to post a new reply.
  */
-function createComment(comment) {
-  const element = document.createElement('li');
-  element.setAttribute(ATTR_ID, comment.commentKey.id);
-
-  const content = document.createElement('span');
-  content.innerText = `${comment.author.email} says ${comment.content}`;
-  element.appendChild(content);
-
-  const repliesDiv = document.createElement('div');
+class DiscussionComment extends HTMLElement {
+  constructor(comment) {
+    super();
+    this.attachShadow({mode: 'open'});
+    const shadow = TEMPLATE_COMMENT.content.cloneNode(true);
+    this.shadowRoot.appendChild(shadow);
+
+    const username = comment.author.email.split('@')[0];
+    const timestampPrefix = comment.parentKey.value ? '' : '00:00 - ';
```

Comment:
Can you add a TODO to update `00:00` with the timestamp once it's available?

Reply:
I moved this into its own function with a TODO, thanks!